concepts/websockets.textile

```diff
@@ -0,0 +1,593 @@
+** "WebSocket libraries you can use right now":#websocket-libraries-you-can-use-right-now
+*** "ws":#ws
+**** "Minimal code samples":#minimal-code-samples
+*** "μWebSockets":#%CE%BCwebsockets
+**** "Minimal code sample":#minimal-code-sample
+*** "faye-websocket":#faye-websocket
+**** "Minimal code samples":#minimal-code-samples
+*** "Socket.io":#socketio
+**** "Minimal code samples":#minimal-code-samples
+*** "SocketCluster":#socketcluster
+**** "Minimal code samples":#minimal-code-samples
+*** "SockJS":#sockjs
+**** "Minimal code samples":#minimal-code-samples
+*** "deepstream server":#deepstream-server
+** "Scaling beyond a single server":#scaling-beyond-a-single-server
+** "Ably: Realtime messaging as a service":#ably-realtime-messaging-as-a-service
+
+h2(#first-a-bit-of-background). First, a bit of background
+
+WebSockets have actually only been generally available for a few years. Before they came along, we had still been able to build robust, essentially "real-time" applications for a while, but it was a lot more painful to do so, and moreso the further back in time we look.
+
+The web has been built on the back of the HTTP protocol, which was originally designed entirely as a request-response mechanism. Open a connection, describe what you want, get back a response, then close the connection. This was fine in the early days of the web because, back then, we were really just dealing with a text document and maybe a few additional assets (usually images).
+
+h3(#scripting-the-web-with-javascript). Scripting the web with JavaScript
+
+In 1995, Netscape Communications hired Brendan Eich with the goal of embedding scripting capabilities into their Netscape Navigator browser, and thus JavaScript was born. Initially JavaScript was kind of weird and couldn't do a whole lot (especially with the extremely limited browser DOM that JavaScript had at its disposal), but it was useful for a few things, such as simple validation of input fields before submitting an HTML form to the server.
+
+Microsoft soon entered the arena with Internet Explorer, which is where the original "browser wars" really began. Both companies were competing with each other to have the best browser, so features and capabilities were inevitably added on a regular basis to both Netscape and Internet Explorer as each company strove to outdo the other.
+
+h3(#the-birth-of-xmlhttprequest-and-ajax). The birth of XMLHttpRequest and AJAX
+
+Two of the most significant capabilities that were soon introduced at that time were the ability to embed Java applets into a page, and Microsoft's own offering - ActiveX controls. These were essentially precompiled components that could optionally present an embedded user interface of their own within a web page. More than that though, they allowed a whole host of additional possibilities beyond JavaScript's own (at the time) meagre suite of scripting capabilities.
```

Comment:
American preferred spelling is 'meager' as per http://grammarist.com/spelling/meager-meagre/, please update